### PR TITLE
shell/report: allow "moreinfo" details to appear in reports; sort benchmark results in reports

### LINKED ISSUE #223: Include "moreinfo" in reports
The moreinfo data should be included in reports, perhaps selectively. Things like the CPU Package information are completely missing from the report because they are in the moreinfo of the "Package Information" item in Devices/Processor.

### PR DESCRIPTION
**Pull request by bp0**
* Fixes #223
* Used by  Devices/Processors/Package Info
* Used by Devices/Device Tree/Summary
* Can be forced for all items with -w

Somewhat related:
* Report benchmark results are now sorted and limited to the 10 nearest results to avoid clutter.
* A new "shell" report format that is very useful for debugging, is not advertised in --help

### CODE REVIEW

File: shell/report.c [hunk 1]
```diff
@@ -34,35 +34,37 @@ static FileTypes file_types[] = {
     {NULL, NULL, NULL, NULL}
 };
 
+/* virtual functions */
 void report_header(ReportContext * ctx)
-{
-    ctx->header(ctx);
-}
+{ ctx->header(ctx); }
```
Comment (lpereira) [inline shell/report.c]: Why the change in indentation?

**Comment by lpereira** [comment]
Nice feature! Will try this in the next few days

File: shell/report.c (hunk 1 above)
Comment (bp0) [reply shell/report.c]: Those bugged me a little, and I was just trying to make that section more compact. I guess report_header(ctx) is better than using ctx->header(ctx) directly? Because of the double ctx maybe. Idk.
